demos/background_subtraction_demo/python/background_subtraction_demo.py

```diff
@@ -176,14 +202,26 @@ def main():
     elif args.adapter == 'ovms':
         model_adapter = OVMSAdapter(args.model)
 
-    labels = ['__background__', 'person'] if args.labels is None else args.labels
+    if args.labels is None:
+        labels = ['__background__', 'person']
+    else:
+        with open(args.labels, 'rt') as labels_file:
+            labels = labels_file.read().splitlines()
+        assert len(labels), 'The file with class labels is empty'
 
     configuration = {
         'confidence_threshold': args.prob_threshold,
         'resize_type': args.resize_type
     }
 
-    model = get_model(model_adapter, configuration)
+    model, is_matting_model, need_bgr_input = get_model(model_adapter, configuration, args.bgr)
+
+    input_bgr = open_images_capture(args.bgr, True) if need_bgr_input else None
+    if not need_bgr_input and args.bgr is not None:
+        log.info('\'--bgr\' argument is set but not needed, so will not be used')
+
+    if args.raw_output_message and is_matting_model:
+        log.info('\'--raw_output_message\' argument is set but is used background-matting based model, nothing to show')
```

Comment:
You should move these warnings into get_model(). With that main() won't need to know the actual model type.

Reply:
Moved inside `get_model` function.